app/index.js

```diff
@@ -60,10 +60,15 @@ function ConnectDialog () {
   self.channelName = ko.observable('')
   self.joinOnly = ko.observable(false)
   self.visible = ko.observable(true)
+  self.loading = ko.observable(false)
   self.show = self.visible.bind(self.visible, true)
   self.hide = self.visible.bind(self.visible, false)
   self.connect = function () {
-    self.hide()
+    self.loading(true)
+    if( !ui.config.defaults.easyMode) {
+      // hide the dialog when loading is done
+      self.loading.subscribe(loading => !loading ? self.hide() : null)
```

Comment:
This `subscribe` call should be outside the `self.connect` function, otherwise it'll add a new subscription on each connection attempt.